pkg/xray/watches.go

```diff
@@ -162,24 +201,77 @@ func packProjectResources(resources WatchProjectResources) []interface{} {
 		if len(res.RepoType) > 0 {
 			resourceMap["repo_type"] = res.RepoType
 		}
-		resourceMap["filter"] = packFilters(res.Filters)
+
+		resourceMap, errors := packFilters(res.Filters, resourceMap)
+		tflog.Error(ctx, fmt.Sprintf(`failed to pack filters: %v`, errors))
+
 		list = append(list, resourceMap)
 	}
 
 	return list
 }
 
-func packFilters(filters []WatchFilter) []interface{} {
-	var l []interface{}
-	for _, f := range filters {
-		m := map[string]interface{}{
-			"type":  f.Type,
-			"value": f.Value,
+type PackFilterFunc func(filter WatchFilter) (map[string]interface{}, error)
+
+func packStringFilter(filter WatchFilter) (map[string]interface{}, error) {
+	var value string
+	err := json.Unmarshal(filter.Value, &value)
+	if err != nil {
+		return nil, err
+	}
+
+	return map[string]interface{}{
+		"type":  filter.Type,
+		"value": value,
+	}, nil
+}
+
+func packAntFilter(filter WatchFilter) (map[string]interface{}, error) {
+	var value WatchFilterAntValue
+	err := json.Unmarshal(filter.Value, &value)
+	if err != nil {
+		return nil, err
+	}
+
+	return map[string]interface{}{
+		"exclude_patterns": value.ExcludePatterns,
+		"include_patterns": value.IncludePatterns,
+	}, nil
+}
+
+func packFilters(filters []WatchFilter, resources map[string]interface{}) (map[string]interface{}, []error) {
+	var stringFilters []map[string]interface{}
+	var antFilters []map[string]interface{}
+	var errors []error
+
+	for _, filter := range filters {
+		var packFilterFunc PackFilterFunc
+		var filtersSlice *[]map[string]interface{}
+
+		switch filter.Type {
+		case "regex", "package-type":
+			packFilterFunc = packStringFilter
+			filtersSlice = &stringFilters
+		case "ant-patterns":
+			packFilterFunc = packAntFilter
+			filtersSlice = &antFilters
+		default:
+			return nil, []error{fmt.Errorf("invalid filter.Type: %s", filter.Type)}
+			break
```

Comment:
Wait... go cases `break` at the next case? I'll go check, but this break here at the bottom looks entirely superflous because you're `return`ing right above

Reply:
oh right! LOL